types/sinon/sinon-tests.ts

```diff
@@ -250,16 +250,16 @@ function testMatch() {
 
 function testFake() {
     const fn = () => { };
-    let fake = sinon.fake();
-
-    fake = sinon.fake(() => true);
-    fake = sinon.fake.returns(5);
-    fake = sinon.fake.throws('foo');
-    fake = sinon.fake.throws(new Error('foo'));
-    fake = sinon.fake.resolves('foo');
-    fake = sinon.fake.rejects('foo');
-    fake = sinon.fake.yields(1, 2, fn);
-    fake = sinon.fake.yieldsAsync(1, 2, fn);
+    const fake = sinon.fake();
+
+    const fakeFn = sinon.fake(() => true);
```

Comment:
these were purposely the same variable before to assert you can assign all the different fakes to the same type

Reply:
good point -- I will have a go at some tests along the lines of your suggestion today